scripts/tests/test_zkai_attention_kv_stwo_fusion_mechanism_ablation_gate.py

```
@@ -0,0 +1,252 @@
+        payload = copy.deepcopy(self.payload)
+        payload["binary_typed_accounting"]["cli_upstream_stwo_serialization_status"] = 1
+        with self.assertRaisesRegex(gate.FusionMechanismAblationGateError, "binary status must be string"):
+            gate._string_payload_field(
+                payload,
+                ("binary_typed_accounting", "cli_upstream_stwo_serialization_status"),
+                "binary status",
+            )
+
+        payload = copy.deepcopy(self.payload)
+        payload["section_delta"]["opening_bucket_savings_share"] = math.inf
+        with self.assertRaisesRegex(gate.FusionMechanismAblationGateError, "section opening share must be finite"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)